### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by fcollonval** [jupyterlab/jupyter-builder#21]
Writing some documentation for this package

---

<details><summary>Raw Markdown</summary>

````
Plan of the doc

- how to install the package?

```sh
pip install ...
```

- what does it do?

  - Provides a cli for building Jupyter extensions. There is 3 subcommands
    - `build`: Build the Jupyter extension JavaScript assets to be consumed by the Jupyter app
    - `develop`: Install the Jupyter extension JavaScript assets in dev mode for consumption in the Jupyter app. It similar to [_editable_ install mode of pip](https://setuptools.pypa.io/en/latest/userguide/development_mode.html).
    - `watch`: Automatically rebuild the development JavaScript assets when one file is changed to ease development.

  - Provides a NPM package manager: `jlpm`
  - [maybe] Provides a NPM package hosting JavaScript asset building configuration (aka `@jupyterlab/builder` see #16)

- how to uninstall the package?

```sh
pip uninstall ...
```

- how to contribute?

> You describe how to install it in development mode and how to test the package
````

</details>

Plan of the doc

- how to install the package?

```sh
pip install ...
```

- what does it do?

  - Provides a cli for building Jupyter extensions. There is 3 subcommands
    - `build`: Build the Jupyter extension JavaScript assets to be consumed by the Jupyter app
    - `develop`: Install the Jupyter extension JavaScript assets in dev mode for consumption in the Jupyter app. It similar to [_editable_ install mode of pip](https://setuptools.pypa.io/en/latest/userguide/development_mode.html).
    - `watch`: Automatically rebuild the development JavaScript assets when one file is changed to ease development.

  - Provides a NPM package manager: `jlpm`
  - [maybe] Provides a NPM package hosting JavaScript asset building configuration (aka `@jupyterlab/builder` see #16)

- how to uninstall the package?

```sh
pip uninstall ...
```

- how to contribute?

> You describe how to install it in development mode and how to test the package

**Comment by fcollonval** [comment]
Markdown tricks for GitHub:

Add links
[text to display](https://github.com/jupyterlab/jupyterlab-builder/pull/11)

Display code snippet :warning: only works in GitHub for issues and pull requests:

https://github.com/jupyterlab/jupyterlab-builder/blob/c04e4a2523df1ccbfc2d4bf9edb34cdd2bd25fb3/jupyter_builder/federated_extensions_requirements.py#L41-L50

> [!Note]
> This a note

> [!WARNING]
> This warning

Link to GitHub documentation: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

**Comment by fcollonval** [comment]
Closed by #22
